interface/Tools.h

```diff
@@ -225,10 +223,26 @@ void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilept
         // di-electron filters: from the path name the legs have symetric cuts
         // meaning: set everything to true, leg1 and leg2 are determined by the highest hlt pt
         if (HH_HLT_DEBUG) std::cout << "\tfinding dilepton legs: di-electron" << std::endl;
-        leptons[dilepton.ilep1].hlt_leg1 = true;
-        leptons[dilepton.ilep1].hlt_leg2 = true;
-        leptons[dilepton.ilep2].hlt_leg1 = true;
-        leptons[dilepton.ilep2].hlt_leg2 = true;
+        filter_leg1.push_back("hltEle17Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg1Filter");
+        filter_leg1.push_back("hltEle23Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg1Filter");
+        filter_leg2.push_back("hltEle17Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg2Filter");
+        filter_leg2.push_back("hltEle23Ele12CaloIdLTrackIdLIsoVLTrackIsoLeg2Filter");
+        for (auto& i1: l1_samepath_indices)
```

Comment:
Only the list of filters differ between double mu and double ele. Can you factorize the matching leg part?

Also, we do not use Ele17Ele12. Is it useful to try to match its filters?